builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Security Considerations
+description: A description of the main differences that Ethereum developers need to understand in terms of security considerations when developing on Moonbeam.
+---
+
+# Security Considerations
+
+![Moonbeam v Ethereum - Security Banner](/images/builders/get-started/eth-compare/security-banner.png)
+
+## Introduction {: #introduction }
+
+When developing smart contracts on Moonbeam, there are some security considerations to be aware of that do not apply when developing on Ethereum. Moonbeam has several [precompiled contracts](/builders/pallets-precompiles/precompiles/){target=_blank}, which are Solidity interfaces that enable developers to access Substrate-based functionality through the Ethereum API. Although the precompiled contracts are designed to improve the developer experience, there can be some unintended consequences that must be considered.
+
+This guide will outline and provide examples of some security considerations to be cognizant of when developing on Moonbeam.
+
+## Arbitrary Code Execution {: #arbitrary-code-execution }
+
+Smart contracts can allow arbitrary code exeuction through low-level message passing to other smart contracts via the [`call()` function](https://solidity-by-example.org/call/){target=_blank}, which is an available method for the [address data type in Solidity](https://docs.soliditylang.org/en/latest/types.html#address){target=_blank}.
+
+The `call()` function accepts ABI encoded call data that is then passed to the target contract. To get the encoded call data, you can use any of the [ABI encoding functions outlined in the Solidity docs](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#abi-encoding-and-decoding-functions){target=_blank}, including `abi.encodeWithSelector` which will be used in the examples in this guide. 
+
+As previously mentioned, one major concern of allowing low-level calls to arbitrarily execute code on Moonbeam is that Moonbeam has precompile contracts that can be called, which can be used to get around some protections that are typically available on Ethereum. To safely use arbitrary code execution on Moonbeam, you should consider the following, which **only applies to contracts that allow arbitrary code execution**:
+
+- Moonbeam [precompiled contracts](builders/pallets-precompiles/precompiles/){target=_blank} such as the Native ERC-20 precompile, XC-20 precompiles, and XCM-related precompiles allow users to manage and transfer assets without requiring access to the EVM. Instead, these actions are done using the Substrate API. So, if your contract holds native tokens or XC-20s and allows arbitrary code execution, these precompiles can be used to drain the balance of the contract
+- Setting a value of `0` when using the `call()` function (for example, `call{value: 0}(...)`) can be overridden by calling the native asset precompile and specifying an amount to transfer in the encoded call data
```

Comment:
There's a small distinction that just occurred to me.

Let contract A be a contract with arbitrary code execution. Let contract B be a contract with a function that has:
`require(msg.value > 1 ether);`

Doing the following will still guarantee that the function on contract B cannot be called by contract A.
`call{value: 0}(...)`

Of course, the funds themselves are still at risk due to the precompile. Not sure if this is enough of an issue to really mention.

Reply:
@chebii2 this is not the right place to look for support. Please contact the team via the official support channels (Telegram/Discord) which you can find in https://moonbeam.network